dependencies.yaml

```diff
@@ -98,26 +98,31 @@ dependencies:
               cuda: "11.2"
             packages:
               - cuda-version=11.2
-              - cudatoolkit
           - matrix:
               cuda: "11.4"
             packages:
               - cuda-version=11.4
-              - cudatoolkit
           - matrix:
               cuda: "11.5"
             packages:
               - cuda-version=11.5
-              - cudatoolkit
           - matrix:
               cuda: "11.8"
             packages:
               - cuda-version=11.8
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - cuda-version=12.0
+      - output_types: conda
+        matrices:
+          - matrix:
+              cuda: "11.*"
+            packages:
               - cudatoolkit
           - matrix:
               cuda: "12.*"
             packages:
-              - cuda-version=12.0
```

Comment:
Should this be dropped entirely?

Reply:
No, we need `cuda-version` to be pinned.

Reply:
Ok am struggling to square this comment with the comment below

If we think we want a pin, why are we not adding a new one (either with my suggestion or a different one)?

Reply:
See my comment on the moved pinning above: https://github.com/rapidsai/ucxx/pull/150/files#r1440878433

Reply:
That isn't related to this question

Put another way, what happens to this environment when CUDA 13 exists?

Reply:
Values of the `cuda` matrix entry must match to an entry in each `specific:` matrix set. The rules for dependency file generation are documented here: https://github.com/rapidsai/dependency-file-generator#how-dependency-lists-are-merged.

If you try to run this with a `cuda` value that isn't in the matrix set above `[11.2, 11.4, 11.5, 11.8, 12.0]` you get an error. In the near future we will be adding support for 12.2 in accordance with https://github.com/rapidsai/build-planning/issues/6, and we'll need to add an entry for `12.2` to have `cuda-version=12.2` in the first list above when we do that.

```
$ rapids-dependency-file-generator --output conda --file_key all --matrix "cuda=12.1;arch=x86_64;py=3.10"
...
ValueError: No matching matrix found in 'cudatoolkit' for: {'cuda': '12.1', 'arch': 'x86_64', 'py': '3.10'}
```

Reply:
Ok thanks Bradley! 🙏